REL-3349: PIT visitor instrument changes.

sraaphorst opened the pull request:
The following have been listed as requirements for the 2018B PIT:

1. Remove TEXES from the list of instruments.
2. Add Phoenix to the list of instruments (GS only).
3. Make DSSI GS only, as Alopeke will be used for GN instead.

In this PR, I adjust the instrument list accordingly, i.e. removing TEXES and adding Phoenix back, and change DSSI to be GS only. Here is the instrument list now:
![screen shot 2018-02-01 at 2 55 59 pm](https://user-images.githubusercontent.com/8795653/35694868-765d3572-0761-11e8-9ff6-8d22039eb173.png)



Here is what you now see when you select DSSI. (This is how Phoenix works, despite the strangeness of a one option selection box, so I followed suit to indicate that it is only at GS, especially important now since this has not always been the case.)
![screen shot 2018-02-01 at 2 56 20 pm](https://user-images.githubusercontent.com/8795653/35694851-6425fccc-0761-11e8-8271-9688ff024a85.png)

The UpConverter now takes DSSI observations at GN and issues a warning, and changes them to GS with the information that Alopeke should be used for GN observations instead.
![screen shot 2018-02-01 at 2 57 33 pm](https://user-images.githubusercontent.com/8795653/35694720-e6d5849a-0760-11e8-851e-089aa1d547ef.png)

Additionally, the UpConverter also assigns Phoenix proposals to GS. This was done in an earlier upconversion, so I simply grabbed the code out of there and moved it to the 2018B conversion.

Finally, the list of problems now generates issues if the configuration includes TEXES, or Phoenix at GN, or DSSI at GN. This should never happen based on the upconversion, but I thought it prudent to include it anyway.

cquiroz commented on bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/transform/UpConverter.scala:
```diff
@@ -138,7 +138,38 @@ case class LastStepConverter(semester: Semester) extends SemesterConverter {
+          case _                         => n
+        }
+      }
+      StepResult("DSSI proposal has been assigned to Gemini South. Use ʻAlopeke instead for Gemini North.", <dssi>{DssiSiteTransformer.transform(ns)}</dssi>).successNel
```
Why would you say `use Alopeke` when converting DSSI?

sraaphorst replied:
From conversations, I was under the impression that Alopeke was being offered as a "better" DSSI, but I might be wrong. I'll double check that with Andy.

sraaphorst replied:
Okay, talking to @andrewwstephens, it looks like Alopeke in speckle mode is an enhanced DSSI, so DSSI and GN proposals should migrate to that instead, with a message being issued. I'll make that change.

sraaphorst commented:
@cquiroz: Now I changed the code so that DSSI GN proposals automatically get changed to Alopeke in speckle mode. I added a test case to the spec file to test this.

cquiroz commented:
Thanks for the fixes
